### PR TITLE
DOC: Improve docs for `jax.numpy`: `arcsinh`, `arccosh` and `arctanh`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/ufuncs.py [hunk 1]
```diff
@@ -817,14 +817,110 @@ def cosh(x: ArrayLike, /) -> Array:
...
 def arcsinh(x: ArrayLike, /) -> Array:
+  r"""Calculate element-wise inverse of hyperbolic sine of input.
+
+  JAX implementation of :obj:`numpy.arcsinh`.
+
+  The inverse of hyperbolic sine is defined by:
+
+  .. math::
+
+    arcsinh(x) = ln(x + \sqrt{1 + x^2})
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:831]: Here and below, within math expressions please use `\ln` in place of `ln`

File: jax/_src/numpy/ufuncs.py [hunk 2]
```diff
@@ -885,9 +981,52 @@ def tanh(x: ArrayLike, /) -> Array:
...
+  Args:
+    x: input array or scalar.
+
+  Returns:
+    An array of same shape as ``x`` containing the inverse of hyperbolic tangent
+    of each element of ``x``, promoting to inexact dtype.
+
+  Note:
+    - ``jnp.arctanh`` returns ``nan`` for real-values outside the range ``[-1, 1]``.
+    - ``jnp.arctanh`` is equivalent to computing ``1j * jnp.arctan(x / 1j)``.
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:1006]: I would remove this: we don't need to note well-known mathematical identities.

Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py:1006]: Should I remove this in `arcsinh` doc also?

Comment (jakevdp) [reply jax/_src/numpy/ufuncs.py:1006]: Yes, I think so

File: jax/_src/numpy/ufuncs.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Done

File: jax/_src/numpy/ufuncs.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Thanks! Removed.

File: jax/_src/numpy/ufuncs.py [hunk 3]
```diff
@@ -817,14 +817,109 @@ def cosh(x: ArrayLike, /) -> Array:
...
+    arcsinh(x) = \ln(x + \sqrt{1 + x^2})
+
+  Args:
+    x: input array or scalar.
+
+  Returns:
+    An array of same shape as ``x`` containing the inverse of hyperbolic sine of
+    each element of ``x``, promoting to inexact dtype.
+
+  Note:
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:840]: Should we add a note about the valid domain, similar to `arccosh` and `arctanh`?

Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py:840]: Thanks! Added a note on valid domain of `arcsinh`.

**Review by jakevdp — APPROVED** [review]
Looks good! Can you squash the commits please?

**Comment by rajasekharporeddy** [comment]
Thank! Squashed into single commit.

**Review by jakevdp — APPROVED** [review]

**Comment by rajasekharporeddy** [comment]
Hi @jakevdp, Fix the undefined variable error. Could you please run the tests once again?

**Pull request merged**